src/plugins/visualize/public/application/components/visualize_editor_common.tsx

```diff
@@ -53,6 +57,57 @@ export const VisualizeEditorCommon = ({
   embeddableId,
   visEditorRef,
 }: VisualizeEditorCommonProps) => {
+  const { services } = useKibana<VisualizeServices>();
+
+  useEffect(() => {
```

Comment:
no tests for this effect. If you move it into separate file it will be easier to cover that by JEST

Reply:
Even here it shouldn't be too complex - tests for graph might be helpful here:
aliasMatch: https://github.com/elastic/kibana/pull/109617/files#diff-200266799564618d6bbdeaceb64cb4d65133d69c5251e30dbc6dc3314e3635abR74
conflict: https://github.com/elastic/kibana/pull/109617/files#diff-cca894dea03ddccd269d0244a9edc54152ac317c069d25424456b4ff30632c8aR49

Reply:
I've already written tests for this src\plugins\visualize\public\application\components\visualize_editor_common.test.tsx